pkg/proxy/iptables/proxier.go

```diff
@@ -175,7 +175,7 @@ func newServiceInfo(serviceName proxy.ServicePortName, port *api.ServicePort, se
 
 	if info.onlyNodeLocalEndpoints {
 		p := apiservice.GetServiceHealthCheckNodePort(service)
-		if p == 0 {
+		if p == 0 && service.Spec.Type == api.ServiceTypeLoadBalancer {
```

Comment:
This isn't right.  You should call NeedsHealthCheck,but even that is wrong, because it doesn't check this condition.  And even if it did it would be wrong, I think, because NodePort services can also set this, which isn't considered.

We probably ALSO need to fix shouldCheckOrAssignHealthCheckNodePort, which also only tests for LoadBalancer.

@MrHohn owns taking this from beta to GA, so I will assign to him